Tools/clinic/clinic.py

```diff
@@ -4840,6 +4840,21 @@ def state_dsl_start(self, line: str) -> None:
 
         self.next(self.state_modulename_name, line)
 
+    def update_function_kind(self, fullname: str) -> None:
```

Comment:
I'm not too wild about the naming here. Let me know if you've got a better suggestion, @alexwaygood. Alternatively, we can split it in two:

- `verify_function_name()`
- `update_function_kind()`

I'm not sure that is worth it, though.

Ideally, this should just be a part of the extracted method in gh-107964